js/views/pageNavVw.js

```diff
@@ -368,16 +371,171 @@ module.exports = Backbone.View.extend({
+
+
+    if($('textarea#aboutInput').val() != ""){
+        self.model.set('about', $('textarea#aboutInput').val());
+    }
+
+    if($('#storeHandleInput').val() != "" && /^@/.test($('#storeHandleInput').val()) ){
+        self.model.set('handle', $('#storeHandleInput').val());
+    }
+
+    if($('#storeNameInput').val() != ""){
+        self.model.set('name', $('#storeNameInput').val());
+    }else if (self.model.get('name') == undefined){
+        //otherwise error since the profile api needs the name parameter and as of now it is not set in the userMd.js
+        self.model.set('name', "");
+    }
+
+    var themeRadioButtons = $('input[name=theme-selection]');
+    var themeSelected =   ($(themeRadioButtons.filter(':checked'))[0] !=undefined);
+    if(themeSelected){
+       var themeId = $(themeRadioButtons.filter(':checked')[0]).attr('id');
+
+        var primaryColor =  parseInt($($("label[for='"+themeId+"']")[0]).data('primary-color').replace("#","0x"));
+        var secondaryColor =   parseInt($($("label[for='"+themeId+"']")[0]).data('secondary-color').replace("#","0x"));
+        var backgroundColor =  parseInt($($("label[for='"+themeId+"']")[0]).data('background-color').replace("#","0x"));
+        var textColor =   parseInt($($("label[for='"+themeId+"']")[0]).data('text-color').replace("#","0x"));
+        var header = $($("label[for='"+themeId+"']")[0]).data('header');
+
+
+        self.model.set('primary_color', primaryColor);
+        self.model.set('secondary_color', secondaryColor);
+        self.model.set('text_color', backgroundColor);
+        self.model.set('background_color', textColor);
+        //TODO upload Image header
+        //From profile api : header= the hash of the header image. must have been previously uploaded using the upload_image api call. (40 character hex string)
+        //self.model.set('header', header);
+
+    }
+
+    $.each(this.model.attributes,
+            function(i,el) {
+                if(i == "country") {
+                    profileFormData.append("location",el);
+                }
+                if(i == "name" || i == "handle" || i =="about"|| (themeSelected && (i == "primary_color" || i == "secondary_color" || i == "text_color"|| i =="background_color" ))) {
+                    profileFormData.append(i,""+el);
+                } else if(i == "tempAvatar") {
+                    var imageURI = $('#avatarUploadDiv').cropit('export', {
+                        type: 'image/jpeg',
+                        quality: 0.75,
+                        originalSize: false
+                    });
+                    if(imageURI) {
+                        imageURI = imageURI.replace(/^data:image\/(png|jpeg);base64,/, "");
+                        uploadImageFormData.append('image', imageURI);
+                    }
+                } else {
+                    settingsFormData.append(i,el);
+                }
+
+            }
+        );
+
+      settingsFormData.append("shipping_addresses","");
```

Comment:
Remove this line, sending a blank address could cause the client to crash on reload. There's code in the current master that deals with that, but it's better not to create bad data.